README.md

```diff
@@ -343,7 +343,56 @@ TODO: description
 
 ### Entry
 
-TODO: Linda
+#### `GET /api/entries` - Get all entries of the user
+
+**Returns**
+- The array of entries by the user sorted in descending order by date
+
+**Throws** 
+- `401` if the user is not logged in
+
+#### `POST /api/entries` - Make a new entry in the log
+
+**Body**
+- `type` _{string}_ - The type of the log entry
+- `detail` _{string}_ - The detail of the log entry
+- `notes` _{string}_ - The log entry's notes (optional)
+- `date` _{Date}_ - The date associated with the log entry (optional)
```

Comment:
Letting date be optional s.t. if a date is not provided (though it shouldn't happen if frontend has default value), today's date is used

Reply:
this seems reasonable (although you're right that it wouldn't be relevant if the frontend has a default value. Should we make this consistent with the trend post?

Reply:
That seems reasonable. Relatedly, do we care about the edge case where the date supplied is non-empty but not a valid date? Presumably that should only happen if a manual (?) POST request is sent without interacting with the frontend?

Reply:
At this moment, I've defined trends in such a way that they depend on dates. Because the frontend has a default value, how could I update the trend post to reflect that?

I think if the date is invalid, we could throw a 400?

Reply:
If the frontend has a default value, I think it should be assumed that API calls should always have a date supplied, _if_ it's made through the frontend?

`400` sounds good. I'm assuming that the date in the body of the request is a string, which is cast to a Date type in the route; and if the casting fails, the date is invalid?

(P.S.: unrelated to API design, but should we also check if date is within a "reasonable" range. whatever that is?)

Reply:
interesting. I pretty much exclusively use [Unix timestamps](https://en.wikipedia.org/wiki/Unix_time) to represent dates. It is a ubiquitous format that only requires a single integer of space and every language I have encountered has built in support for formatting in a human-readable way.

Reply:
also only having an integer means we don't have to do any parsing or more complicated error checking that we could potentially mess up (daylight savings, leap days/hours, etc.). Generally messing around with human time constructs is a huge pain.

Reply:
As far as a "reasonable" range there are I guess two things to consider. (1) what inputs could break the correctness of the backend. This is partially a security concern and we should consider all inputs to the API in isolation from the frontend. My guess is that if we use timestamps we can be relatively confident that we won't have issues with this. (2) what inputs could negatively impact the user experience. This is more about accounting for users' mistakes. For this category we can assume well-intentioned users, which implies they will be using the front end. If we have concerns about users entering a date in the future for example, one option is to warn them on the front end (or not even give them the ability to select that date in the first place) and never even send to the back end. Someone with bad intentions could get around this check but if the impact is just a worse experience for them and doesn't impact the integrity of the server or other users' data then it probably doesn't matter.